src/providers/pkcs11_provider/key_metadata.rs

```diff
@@ -0,0 +1,115 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use super::{KeyInfo, KeyMetadata, Pkcs11Provider};
+use crate::key_info_managers;
+use crate::key_info_managers::KeyTriple;
+use log::{error, warn};
+use parsec_interface::operations::psa_key_attributes::*;
+use parsec_interface::requests::{ResponseStatus, Result};
+
+impl Pkcs11Provider {
+    /// Gets a key identifier and key attributes from the Key Info Manager.
+    pub(super) fn get_key_info(&self, key_triple: &KeyTriple) -> Result<([u8; 4], Attributes)> {
```

Comment:
Could this method return a `KeyMetadata`?

Reply:
Do you think we should have `KeyMetadata` and use that everywhere, or just drop that struct?

Reply:
oh yeah maybe drop it then if it's not needed anymore